### PR TITLE
Convert MediaTag to a view component

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: app/components/embed/media_component.rb [hunk 1]
```diff
@@ -7,5 +7,22 @@ def initialize(viewer:)
...
+      Settings.enabled_features.transcripts || params[:transcripts] == 'true'
+    end
+
+    def primary_files_count
+      purl_object.contents.count do |resource|
+        resource.primary_file.present?
+      end
+    end
+
+    def many_primary_files
```
Comment (mjgiarlo) [inline app/components/embed/media_component.rb:24]:
```suggestion
    def many_primary_files?
```

Comment (jcoyne) [reply app/components/embed/media_component.rb:24]: But then we can't pass it as an implicit argument `many_primary_files:`

**Review by mjgiarlo — APPROVED** [review]
Other than a trivial suggestion, approving.

**Review by mjgiarlo — APPROVED** [review]
Other than a trivial suggestion, approving.

**Pull request merged**